src/commands/lcs.json

```
@@ -0,0 +1,71 @@
+{
+    "LCS": {
+        "summary": "Find longest common substring",
+        "complexity": "O(N*M) where N and M are the lengths of s1 and s2, respectively",
+        "group": "string",
+        "since": "7.0.0",
+        "arity": -3,
+        "function": "lcsCommand",
+        "command_flags": [
+            "READONLY"
+        ],
+        "acl_categories": [
+            "STRING"
+        ],
+        "key_specs": [
+            {
+                "flags": [
+                    "READ"
+                ],
+                "begin_search": {
+                    "index": {
+                        "pos": 1
+                    }
+                },
+                "find_keys": {
+                    "range": {
+                        "lastkey": 1,
+                        "step": 1,
+                        "limit": 0
+                    }
+                }
+            }
+        ],
+        "arguments": [
+            {
+                "name": "key1",
+                "type": "key",
+                "key_spec_index": 0
+            },
+            {
+                "name": "key2",
+                "type": "key",
+                "key_spec_index": 1
```

```suggestion
                "key_spec_index": 0
```
@guybe7  Please confirm it should be 0 here?

---

@yangbodong22011 you're right. there's only one key-spec, so it can't be at index 1.
please make a PR.

p.s. how did you find that? maybe we can benefit from some mechanism that validates there.

---

@oranagra Just found out while looking at the code, I haven't checked the rest of the commands.

---

please make a PR.
------
We might have to think of other ways to make sure other commands don't go wrong, rather than fixing one by one.

---

thanks, @yangbodong22011 will you create a PR, or should I?

and yes, we need to figure out a better way to spot these issues... got any ideas?

---

Idea: One of the python scripts for handling the JSON files could do some validation.

---

Is it possible to check whether the index is valid in generate-command-code.py and whether the key_spec contains an unused index.

---

> Is it possible to check whether the index is valid in generate-command-code.py and whether the key_spec contains an unused index.

i write a small ugly test script, and find out these, you guys can check on it, in case you guys forget this one

- LCS: key2 key_spec_index error
- RENAME: have unused key_spec
- RENAMENX: have unused key_spec

- WATCH: key_specs missing flags?
- SUNSUBSCRIBE / SPUBLISH / SSUBSCRIBE are NOT_KEY, i suppose they are ok
- RESTORE-ASKING: missing arguments, should it be like RESTORE?
- PFDEBUG: missing arguments, this is a debug command, so i suppose it is ok

---

@enjoy-binbin How "ugly" is your script? I think we need it in a test case or a separate CI job.

---

@enjoy-binbin thank you for reminding us.. looks like we forgot it.
so first thing's first, let's fix them, please make a PR.

if your test script is tcl based, let's put it in the test suite.
alternatively, we could set some assertions in the C code.
or put another test in the utils folder and somehow run it in some CI or manually from time to time (not preferred)

---

i just write some logic in generate-command-code and the check spec_key logic base on my understanding
i just wanted to see if we have any other commands that have the same problem first (we can fix it first and then try to find a way to check it)
ok, i can summery it and then let's discuss it in other thread.